pkg/cmd/browse/browse.go

```diff
@@ -0,0 +1,245 @@
+	branchName, _ := api.RepoDefaultBranch(apiClient, baseRepo)
+
+	if !inRepo(err) {
+		return printExit(exitNotInRepo, cmd, opts, "")
+	}
+
+	if getFlagAmount(cmd) > 1 {
+		return printExit(exitTooManyFlags, cmd, opts, "")
+	}
+
+	repoUrl := ghrepo.GenerateRepoURL(baseRepo, "")
+	response := exitUrlSuccess
+
+	if !hasArg(opts) && hasFlag(cmd) {
+		response, repoUrl = addFlag(opts, repoUrl)
+	} else if hasArg(opts) && !hasFlag(cmd) {
+		response, repoUrl = addArg(opts, repoUrl, branchName)
+	} else if hasArg(opts) && hasFlag(cmd) {
+		response, repoUrl = addCombined(opts, repoUrl, branchName)
+	}
+
+	if response == exitUrlSuccess || response == exitNonUrlSuccess {
+		opts.Browser.Browse(repoUrl)
+	}
+
+	return printExit(response, cmd, opts, repoUrl)
+}
+
+func addCombined(opts *BrowseOptions, url string, branchName string) (exitCode, string) {
+
+	if !opts.BranchFlag {
+		return exitInvalidCombo, ""
+	}
+
+	if opts.AdditionalArg == "" {
+		return exitUrlSuccess, url + "/tree/" + opts.SelectorArg
+	}
+
+	arr := parseFileArg(opts)
+	if len(arr) > 1 {
+		return exitUrlSuccess, url + "/tree/" + opts.AdditionalArg + "/" + arr[0] + "#L" + arr[1]
+	}
+
+	return exitUrlSuccess, url + "/tree/" + opts.AdditionalArg + "/" + arr[0]
+}
+
+func addFlag(opts *BrowseOptions, url string) (exitCode, string) {
+	if opts.ProjectsFlag {
+		return exitUrlSuccess, url + "/projects"
+	} else if opts.SettingsFlag {
+		return exitUrlSuccess, url + "/settings"
+	} else if opts.WikiFlag {
+		return exitUrlSuccess, url + "/wiki"
+	}
+	return exitExpectedArg, ""
+}
+
+func addArg(opts *BrowseOptions, url string, branchName string) (exitCode, string) {
+
+	if opts.AdditionalArg != "" {
+		return exitTooManyArgs, ""
+	}
+
+	if isNumber(opts.SelectorArg) {
+		url += "/issues/" + opts.SelectorArg
+		return exitNonUrlSuccess, url
+	}
+
+	arr := parseFileArg(opts)
+	if len(arr) > 1 {
+		return exitUrlSuccess, url + "/tree/" + branchName + "/" + arr[0] + "#L" + arr[1]
+	}
+
+	return exitUrlSuccess, url + "/tree/" + branchName + "/" + arr[0]
+}
+
+func parseFileArg(opts *BrowseOptions) []string {
+	arr := strings.Split(opts.SelectorArg, ":")
+	return arr
+}
+
+func printExit(exit exitCode, cmd *cobra.Command, opts *BrowseOptions, url string) error {
+	w := opts.IO.ErrOut
+	cs := opts.IO.ColorScheme()
+	help := "Use 'gh browse --help' for more information about browse\n"
+
+	switch exit {
+	case exitUrlSuccess:
+		//fmt.Fprintf(w, "now opening %s in browser . . .\n", cs.Bold(url))
+		fmt.Fprintln(w, "Hello World One")
+	case exitNonUrlSuccess:
+		//fmt.Fprintf(w, "now opening issue/pr in browser . . .\n")
+		fmt.Fprintln(w, "Hello World Two")
+	case exitNotInRepo:
+		return fmt.Errorf("change directory to a repository to open in browser\n%s", help)
+	case exitTooManyFlags:
+		return fmt.Errorf("accepts 1 flag, %d flag(s) were recieved\n%s", getFlagAmount(cmd), help)
+	case exitTooManyArgs:
+		return fmt.Errorf("accepts 1 arg, 2 arg(s) were received \n%s", help)
```

Comment:
This validation is already handled for you courtesy of Cobra's `Args:  cobra.RangeArgs(...)`, so you don't have to manually implement the check here.